### PR TITLE
WebGPURenderer: invert viewport/scissor y coords

### PR DESCRIPTION
**Pull request by aardgoose**
WebGPU uses the top left corner of the framebuffer origin for viewport and scissor operations while WebGL uses the bottom left corner. Preserve the existing API by inverting y coordinate and adjust line2 example.

### CODE REVIEW

File: examples/jsm/renderers/webgpu/WebGPUBackend.js [hunk 1]
```diff
@@ -342,15 +347,15 @@ class WebGPUBackend extends Backend {
 
 		if ( renderContext.viewport ) {
 
-			this.updateViewport( renderContext );
+			this.updateViewport( renderContext, bufferHeight );
```
Comment (sunag) [inline examples/jsm/renderers/webgpu/WebGPUBackend.js:350]: I think we can add the properties `height` and `width` in `RenderContext` and remove `bufferHeight`?

Comment (aardgoose) [reply examples/jsm/renderers/webgpu/WebGPUBackend.js:350]: I'll take a look.

File: examples/jsm/renderers/webgpu/WebGPUBackend.js [hunk 2]
```diff
@@ -231,6 +232,9 @@ class WebGPUBackend extends Backend {
 
 			}
 
+			renderContext.height = Math.floor( textures[ 0 ].source.data.height );
```
Comment (sunag) [inline examples/jsm/renderers/webgpu/WebGPUBackend.js:235]: It's possible define `renderContext.width` and `height`  in `Renderer.js`?

**Comment by sunag** [comment]
Thanks!

**Pull request merged**